File: backend/app/schemas/prescription.py
from pydantic import BaseModel
from datetime import datetime


class PrescriptionBase(BaseModel):
    medication_name: str
    dosage: str | None = None
    frequency: str | None = None
    duration: str | None = None
    notes: str | None = None


class PrescriptionCreate(PrescriptionBase):
    pass


class PrescriptionUpdate(BaseModel):
    medication_name: str | None = None
    dosage: str | None = None
    frequency: str | None = None
    duration: str | None = None
    notes: str | None = None


class PrescriptionSchema(PrescriptionBase):
    id: int
    patient_id: int
    prescribed_by: int | None
    prescribed_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/prescription.ts
export interface Prescription {
  id: number;
  patient_id: number;
  medication_name: string;
  dosage: string;
  frequency: string;
  duration: string;
  notes?: string;
  prescribed_by: number;
  prescribed_at: string;
}

export interface PrescriptionCreate {
  medication_name: string;
  dosage: string;
  frequency: string;
  duration: string;
  notes?: string;
}

export interface PrescriptionUpdate {
  medication_name?: string;
  dosage?: string;
  frequency?: string;
  duration?: string;
  notes?: string;
}
